self-core-impl/src/main/java/com/selfxdsd/core/managers/AuthorHasRoles.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2020, Self XDSD Contributors
+ * All rights reserved.
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to read the Software only. Permission is hereby NOT GRANTED to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software.
+ * <p>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core.managers;
+
+import com.selfxdsd.api.*;
+import com.selfxdsd.api.pm.PreconditionCheck;
+import com.selfxdsd.api.pm.Step;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Check if the comment's author has one of the provided roles to perform
+ * the next step.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.20
+ */
+public final class AuthorHasRoles extends PreconditionCheck {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        AuthorHasRoles.class
+    );
+
+    /**
+     * Accepted roles.
+     */
+    private final List<String> acceptedRoles;
+
+    /**
+     * Ctor.
+     * @param onTrue Step to follow if the author has on of the given roles.
+     * @param onFalse Step to follow if the author none of the given roles.
+     * @param roles Accepting roles.
+     */
+    public AuthorHasRoles(final Step onTrue,
+                          final Step onFalse,
+                          final String...roles) {
+        super(onTrue, onFalse);
+        this.acceptedRoles = new ArrayList<>(Arrays.asList(roles));
+    }
+
+    @Override
+    public void perform(final Event event) {
+        final Project project = event.project();
+        final String author = event.comment().author();
+        final Contributor contributor = project
+            .contributors()
+            .getById(author, project.provider());
+        if (contributor == null) {
```

Comment:
is `project.owner()` implicit a contributor? 
Else I should check to see if author is project.owner() and has the role.

Reply:
@criske The owner is not by default a contributor. I think we will make him a Contributor with role ``PO`` (Project Owner), but it's irrelevant for the code here.

When we register a Project, we could call ``Contributors.register(...)`` to add him as a contributor. So don't worry.